common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java

```diff
@@ -53,19 +58,40 @@ public void onTitleScreenInit(TitleScreenInitEvent.Post e) {
     public void onTitleScreenInit(ScreenInitEvent e) {
         if (!(e.getScreen() instanceof TitleScreen titleScreen)) return;
 
-        addWynncraftButton(titleScreen);
+        if (!initBootPassed && autoConnect.get()) {
+            autoConnect();
+        } else {
+            addWynncraftButton(titleScreen);
+        }
+        initBootPassed = true;
     }
 
     private void addWynncraftButton(TitleScreen titleScreen) {
-        ServerData wynncraftServer =
-                new ServerData("Wynncraft", connectToLobby.get() ? LOBBY_SERVER : GAME_SERVER, false);
-        wynncraftServer.setResourcePackStatus(ServerData.ServerPackStatus.ENABLED);
+        ServerData wynncraftServer = getWynncraftServer();
 
         WynncraftButton wynncraftButton = new WynncraftButton(
                 titleScreen, wynncraftServer, titleScreen.width / 2 + 104, titleScreen.height / 4 + 48 + 24);
         titleScreen.addRenderableWidget(wynncraftButton);
     }
 
+    private void autoConnect() {
+        ServerData wynncraftServer = getWynncraftServer();
+
+        ConnectScreen.startConnecting(
```

Comment:
This is shared with the `onPress` method in the button class. Please extract it to a common private method.

And after doing that, `autoConnect` becomes only two lines, and is not needed (can be inlined in place).